internal/dispatch/graph/computecheck_test.go

```diff
@@ -369,6 +371,373 @@ func TestComputeCheckWithCaveats(t *testing.T) {
+					"",
+					nil,
+				},
+			},
+			[]check{
+				{
+					"resource:foo#tag@actor:johndoe",
+					map[string]any{
+						"actor_created_at": "2022-01-01T11:00:00.021Z",
+					},
+					v1.ResourceCheckResult_MEMBER,
+					nil,
+					"",
+				},
+				{
+					"resource:foo#tag@actor:johndoe",
+					map[string]any{
+						"actor_created_at": "2022-01-01T09:00:00.021Z",
+					},
+					v1.ResourceCheckResult_NOT_MEMBER,
+					nil,
+					"",
+				},
+			},
+		},
+		{
+			"time-bound permission",
+			`definition resource {
+						relation reader: user
+						permission view = reader
+					}
+					
+					definition user {}`,
+			map[string]caveatDefinition{
+				"expired": {
+					"now() < timestamp(expiration)",
+					map[string]caveats.VariableType{
+						"expiration": caveats.StringType,
+					},
+				},
+			},
+			[]caveatedUpdate{
+				{
+					core.RelationTupleUpdate_CREATE,
+					"resource:foo#reader@user:sarah",
+					"expired",
+					map[string]any{
+						"expiration": "2030-01-01T10:00:00.021Z",
+					},
+				},
+				{
+					core.RelationTupleUpdate_CREATE,
+					"resource:foo#reader@user:john",
+					"expired",
+					map[string]any{
+						"expiration": "2020-01-01T10:00:00.021Z",
+					},
+				},
+			},
+			[]check{
+				{
+					"resource:foo#view@user:sarah",
+					nil,
+					v1.ResourceCheckResult_MEMBER,
+					nil,
+					"",
+				},
+				{
+					"resource:foo#view@user:john",
+					nil,
+					v1.ResourceCheckResult_NOT_MEMBER,
+					nil,
+					"",
+				},
+			},
+		},
+		{
+			"legal-guardian example",
+			`definition claim {
+					  relation claimer: user
+					  relation dependent_of: user#dependent_of
+					  
+					  permission view = claimer + dependent_of
+					}
+					
+					definition user {
+						  relation dependent_of: user
+					}`,
+			map[string]caveatDefinition{
+				"legal_guardian": {
+					"age < 12 || (class != \"sensitive\" && age > 12 && age < 18)",
+					map[string]caveats.VariableType{
+						"age":   caveats.IntType,
+						"class": caveats.StringType,
+					},
+				},
+			},
+			[]caveatedUpdate{
+				{
+					core.RelationTupleUpdate_CREATE,
+					"user:son#dependent_of@user:father",
+					"",
+					nil,
+				},
+				{
+					core.RelationTupleUpdate_CREATE,
+					"claim:broken_leg#dependent_of@user:son#dependent_of",
+					"legal_guardian",
+					map[string]any{
+						"age":   10,
+						"class": "non-sensitive",
+					},
+				},
+				{
+					core.RelationTupleUpdate_CREATE,
+					"user:daughter#dependent_of@user:father",
+					"",
+					nil,
+				},
+				{
+					core.RelationTupleUpdate_CREATE,
+					"claim:broken_arm#dependent_of@user:daughter#dependent_of",
+					"legal_guardian",
+					map[string]any{
+						"age":   14,
+						"class": "non-sensitive",
+					},
+				},
+				{
+					core.RelationTupleUpdate_CREATE,
+					"claim:sensitive_matter#dependent_of@user:daughter#dependent_of",
+					"legal_guardian",
+					map[string]any{
+						"age":   14,
+						"class": "sensitive",
```

Comment:
I wonder if we should add an enum type of some kind

Reply:
Maybe be reasonable when we get to see the UX with schema parser in place